### PR TITLE
Change poe to nox

### LINKED ISSUE #23: Change Poe to Nox
Currently we use PoeThePoet. We want to replace this with Nox instead.

### PR DESCRIPTION
**Pull request by Nicoretti**
Fixes #23 

- Add nox support to workspace
- Remove poe task executer from workspace
- Remove doc scripts

### CODE REVIEW

File: noxfile.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+
+@nox.session(python=False)
+def tests(session):
+    """Run all unit tests"""
+    session.run("pytest", f"{PROJECT_ROOT / 'tests'}")
+
+
+@nox.session(python=False)
+@nox.parametrize("branch", ["current", "main"])
```
Comment (tkilias) [inline noxfile.py:92]: I am not sure, if we need this parameterized

File: noxfile.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        "commit",
+        "--target_branch",
+        target_branch[branch],
+    ]
+    args += branch_specific_arguments[branch]
+    session.run(*args, external=True)
+
+
+@nox.session(python=False)
+@nox.parametrize("target", ["current", "main", "release"])
```
Comment (tkilias) [inline noxfile.py:122]: I am not sure, if we need this parameterized

File: noxfile.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+            "-o",
+            "api",
+            f"{PACKAGE.resolve()}",
+            external=True,
+        )
+        session.run("sphinx-build", "-b", "html", "-W", ".", "build", external=True)
+
+
+@nox.session(python=False)
+def open(session):
```
Comment (tkilias) [inline noxfile.py:73]: probably, not concrete enough

File: noxfile.py [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+def clean(session):
+    """Remove all build and documentation artifacts"""
+    build_directory = DOC / "build"
+    if build_directory.exists():
+        rmtree(build_directory.resolve())
+        session.log(f"Removed {build_directory}")
+
+
+@nox.session(python=False)
+def build(session: nox.session):
```
Comment (tkilias) [inline noxfile.py:57]: probably, not concrete enough

File: noxfile.py [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+    - the code is executed where the working directory is within a git repository
+    """
+    command = ["git", "branch", "--show-current"]
+    result = run(command, stdout=PIPE, stderr=PIPE, check=True)
+    branch_name = result.stdout.decode("utf-8").strip()
+    return branch_name
+
+
+@nox.session(python=False)
+def clean(session):
```
Comment (tkilias) [inline noxfile.py:48]: probably, not concrete enough

**Review by tkilias — CHANGES_REQUESTED** [review]

File: noxfile.py (hunk 1 above)
Comment (Nicoretti) [reply noxfile.py:92]: I think it it reasonable because the 2 scripts it replaces only vary in a few parameters

File: noxfile.py (hunk 2 above)
Comment (Nicoretti) [reply noxfile.py:122]: I think it it reasonable because the 3 scripts it replaces only vary in a few parameters

File: noxfile.py (hunk 1 above)
Comment (tkilias) [reply noxfile.py:92]: ah, I missed something, you already restricted the values to current and main, I thought, the branch would be freely choosable. Then it should be fine.

**Review by tkilias — CHANGES_REQUESTED** [review]

File: noxfile.py [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+    command = ["git", "branch", "--show-current"]
+    result = run(command, stdout=PIPE, stderr=PIPE, check=True)
+    branch_name = result.stdout.decode("utf-8").strip()
+    return branch_name
+
+
+@nox.session(python=False, name='clean-docs')
+def clean(session):
+    """Remove all documentation artifacts"""
+    build_directory = DOC / "build"
```
Comment (tkilias) [inline noxfile.py:50]: Use constant

File: noxfile.py [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+        session.run(
+            "sphinx-apidoc",
+            "-T",
+            "-e",
+            "-o",
+            "api",
+            f"{PACKAGE.resolve()}",
+            external=True,
+        )
+        session.run("sphinx-build", "-b", "html", "-W", ".", "build", external=True)
```
Comment (tkilias) [inline noxfile.py:69]: Make build to constant

File: noxfile.py [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+
+
+@nox.session(python=False, name='open-docs')
+def open(session):
+    """Open the documentation in the browser"""
+    index_page = DOC / "build" / "index.html"
```
Comment (tkilias) [inline noxfile.py:75]: Use constant for build directory

File: noxfile.py [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+    if not index_page.exists():
+        session.error(
+            (f"File {index_page} does not exist." "Please run `nox -s build` first")
+        )
+    session.run(
+        "python", "-m", "webbrowser", "-t", f"{index_page.resolve()}", external=True
```
Comment (tkilias) [inline noxfile.py:81]: Use Webbrowser in Python and not as subprocess

**Review by tkilias — CHANGES_REQUESTED** [review]

**Review by tkilias — APPROVED** [review]

**Pull request merged**